src/utils/keyAndAdresses.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  base58Decode,
+  base58Encode,
+  varintEncode,
+  varintDecode,
+  hashBlake3,
+} from './Xbqcrypto';
+
+import { getBytesSecretKey, getBytesPublicKey } from './bytes';
+
+/**
+ * Prefixes for secret and public keys.
+ * Prefixes are used as a convention to differentiate one key from another.
+ */
+
+const PUBLIC_KEY_PREFIX = 'P';
+const ADDRESS_PREFIX = 'AU';
+const SECRET_KEY_PREFIX = 'S';
+
+/**
+ * A secret key.
+ *
+ * @remarks the secret key object is created from a base58 encoded string representing the secret key.
+ * The secret key bytes representation is `version + secretKey` where secretKey is the 32 bytes of the secret key.
+ */
+export class SecretKey {
+  base58Encoded: string;
+  base58Decoded: Uint8Array;
+  base58DecodedUnversioned: Uint8Array;
```

Comment:
What is behing this concept of base58DecodedUnversioned? Because by default base58 has no version, right?